dashboard/src/modules/components/SidebarComponent/index.jsx

```diff
@@ -8,11 +8,27 @@ import {
 import React, { useEffect } from "react";
 import { menuOptions, menuOptionsNonLoggedIn } from "./sideMenuOptions";
 import { useDispatch, useSelector } from "react-redux";
-import { useLocation, useNavigate } from "react-router-dom";
+import { useLocation, useNavigate, useOutletContext } from "react-router-dom";
 
 import Cookies from "js-cookie";
 import { setActiveItem } from "actions/sideBarActions";
 
+const MenuItem = (props) => {
+  const { data } = props;
+  const navigate = useOutletContext();
+  return data.map((option) => (
+    <NavItem
+      preventDefault
+      onClick={() => navigate(option.link)}
+      key={option.key}
+      itemId={option.key}
+      isActive={props.activeItem === option.key}
+    >
```

Comment:
As long as you are decomposing `props` at line 17, consider pulling `activeItem` out of it as well and save a few characters at line 25.  That is, if I understand things, something like this should work:
```
  const { data, activeItem } = props;
...
      isActive={activeItem === option.key}
```
(That is, it would be best to be consistent about how you access the fields in `props` -- either decompose `props` into a pair of local variables, or access the items using field references, but don't do one one way and one the other.)

Reply:
Actually, I think you can take this one step further by omitting line 17 and changing line 16 to something like:
```
const MenuItem = ({ data, activeItem }) => {
```

Reply:
Got it and updated